### PR TITLE
#7 allura conformance

### PR DESCRIPTION
**Pull request by MaxHellstrom**
- Solves #7 
- added function add a custom normalization_settings file

### CODE REVIEW

File: src/pyskindose/analyze_data.py [hunk 1]
```diff
@@ -1,34 +1,39 @@
...
-        output dictionary containing calculation specifics such as dose map,
-        correction factors, etc..
+        Returns
+        -------
+        Dict[str, Any]
+            output dictionary containing calculation specifics such as dose
+            map, correction factors, etc..
+
+        """
+
```
Comment (BwKodex) [inline src/pyskindose/analyze_data.py:34]: Remove empty line

File: src/pyskindose/geom_calc.py [hunk 2]
```diff
@@ -45,7 +45,17 @@ def calculate_field_size(field_size_mode, data_parsed, data_norm):
         FS_lat = round(100 * np.sqrt(data_parsed.CollimatedFieldArea_m2), 3)
         FS_long = FS_lat
 
-    return FS_lat, FS_long
+        return FS_lat, FS_long
+
+    if field_size_mode == 'ASD':
```
Comment (BwKodex) [inline src/pyskindose/geom_calc.py:50]: Create a constant for 'ASD'

File: src/pyskindose/geom_calc.py [hunk 3]
```diff
@@ -45,7 +45,17 @@ def calculate_field_size(field_size_mode, data_parsed, data_norm):
...
+
+        FS_long = (data_parsed.LeftShutter_mm + data_parsed.RightShutter_mm) / 10
```
Comment (BwKodex) [inline src/pyskindose/geom_calc.py:52]: Maybe we should add some comments (or preferably an aptly named helper function) to make it clear why this is done. A helper function could, e.g., be named `convert_length_measurement_unit_to_cm(measurement_value: float, unit: str)`.

File: src/pyskindose/geom_calc.py [hunk 4]
```diff
@@ -45,7 +45,17 @@ def calculate_field_size(field_size_mode, data_parsed, data_norm):
...
+        FS_lat = (data_parsed.TopShutter_mm + data_parsed.BottomShutter_mm) / 10
+
+        scale = data_norm.DSD / 100
```
Comment (BwKodex) [inline src/pyskindose/geom_calc.py:55]: Try to clarify what is happening here, perhaps using the same kind of solution as I suggested above

File: src/pyskindose/settings_pyskindose.py [hunk 5]
```diff
@@ -351,3 +352,16 @@ def create_attrs_str(
...
 
     return attrs_str
+
+
+def initialize_settings(settings: Union[str, dict, PyskindoseSettings]) -> PyskindoseSettings:
+    valid_input_settings = settings is not None and isinstance(settings, (str, dict, PyskindoseSettings))
+
+    if not valid_input_settings:
+        raise ValueError("Settings must be given as a str or dict")
+
```
Comment (BwKodex) [inline src/pyskindose/settings_pyskindose.py:362]: Remove superfluous empty line

Comment (MaxHellstrom) [reply src/pyskindose/settings_pyskindose.py:362]: fixed in 09d0a67660f2e07f75547e611b5c52b067246d7e

File: src/pyskindose/geom_calc.py (hunk 2 above)
Comment (MaxHellstrom) [reply src/pyskindose/geom_calc.py:50]: fixed

File: src/pyskindose/geom_calc.py (hunk 3 above)
Comment (MaxHellstrom) [reply src/pyskindose/geom_calc.py:52]: Agree.

The function has been rebuilt.

File: src/pyskindose/geom_calc.py (hunk 4 above)
Comment (MaxHellstrom) [reply src/pyskindose/geom_calc.py:55]: same sol as above

File: src/pyskindose/constants.py [hunk 6]
```diff
@@ -26,6 +26,7 @@
 
 DATA_DS_IRP = "DSIRP"
 
+
```
Comment (BwKodex) [inline src/pyskindose/constants.py:29]: Unnecessary blank line

File: src/pyskindose/geom_calc.py [hunk 7]
```diff
@@ -27,25 +27,57 @@ def calculate_field_size(field_size_mode, data_parsed, data_norm):
...
+        i.e. output of function rdsr_parser
+    data_norm : pd.DataFrame
+        RDSR data, normalized for compliance with PySkinDose.
 
     Returns
     -------
-    [type]
-        [description]
+    FS_lat, FS_long : float
+        Field size in lat- and long direction in cm at the detector plane.
```
Comment (BwKodex) [inline src/pyskindose/geom_calc.py:40]: Should there be a blank line here?

**Review by BwKodex — APPROVED** [review]

**Pull request merged**